core/src/main/scala/org/apache/spark/sql/catalyst/catalog/TiCompositeSessionCatalog.scala

```diff
@@ -0,0 +1,344 @@
+                               newStats: Option[CatalogStatistics]): Unit =
+    catalogOf(identifier.database)
+      .getOrElse(
+        throw new NoSuchDatabaseException(identifier.database.getOrElse(getCurrentDatabase))
+      )
+      .alterTableStats(identifier, newStats)
+
+  override def tableExists(name: TableIdentifier): Boolean =
+    catalogOf(name.database)
+      .map {
+        // Need to exclude tables from Ti's default db.
+        case tiCatalog: TiSessionCatalog =>
+          !name.database.getOrElse(getCurrentDatabase).equals("default") && tiCatalog.tableExists(
+            name
+          )
+        case catalog: SessionCatalog => catalog.tableExists(name)
+      }
+      .getOrElse(throw new NoSuchDatabaseException(name.database.getOrElse(getCurrentDatabase)))
+
+  override def getTableMetadata(name: TableIdentifier): CatalogTable =
+    catalogOf(name.database)
+      .getOrElse(throw new NoSuchDatabaseException(name.database.getOrElse(getCurrentDatabase)))
+      .getTableMetadata(name)
+
+  override def renameTable(oldName: TableIdentifier, newName: TableIdentifier): Unit = {
+    val db = formatDatabaseName(oldName.database.getOrElse(getCurrentDatabase))
+    newName.database.map(formatDatabaseName).foreach { newDb =>
+      if (db != newDb) {
+        throw new AnalysisException(
+          s"RENAME TABLE source and destination databases do not match: '$db' != '$newDb'"
+        )
+      }
+    }
+    catalogOf(oldName.database)
+      .getOrElse(throw new NoSuchDatabaseException(oldName.database.getOrElse(getCurrentDatabase)))
+      .renameTable(oldName, newName)
+  }
+
+  override def dropTable(name: TableIdentifier, ignoreIfNotExists: Boolean, purge: Boolean): Unit =
+    catalogOf(name.database)
+      .getOrElse(throw new NoSuchDatabaseException(name.database.getOrElse(getCurrentDatabase)))
+      .dropTable(name, ignoreIfNotExists, purge)
+
+  override def lookupRelation(name: TableIdentifier): LogicalPlan =
+    catalogOf(name.database)
+      .getOrElse(throw new NoSuchDatabaseException(name.database.getOrElse(getCurrentDatabase)))
+      .lookupRelation(name)
+
+  override def listTables(db: String): Seq[TableIdentifier] = listTables(db, "*")
+
+  override def listTables(db: String, pattern: String): Seq[TableIdentifier] = {
+    val currentSessionCatalog = catalogOf(Some(db)).getOrElse(throw new NoSuchDatabaseException(db))
+    val tables = currentSessionCatalog.listTables(db, pattern)
+    // list tempViews if catalog matches CH Catalog
+    val extraLocalTempViews = currentSessionCatalog match {
+      case _: TiConcreteSessionCatalog =>
+        legacyCatalog
+          .listTables(legacyCatalog.getCurrentDatabase, pattern)
+          .filter(_.database.isEmpty)
+      case _ => Seq()
+    }
+    tables ++ extraLocalTempViews
+  }
+
+  // Following are all routed to primary catalog.
+  override def createDatabase(dbDefinition: CatalogDatabase, ignoreIfExists: Boolean): Unit =
```

Comment:
For TiSpark, it for now only make sense to create table on legacy catalog. Either you hardcode it, or you delete the other policy.

Reply:
I think this makes sense, the other policy is not needed for now.